src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/MonitorProjectConfigurationFilePathEndpointTest.cs

```diff
@@ -0,0 +1,310 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis.Razor;
+using Moq;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    public class MonitorProjectConfigurationFilePathEndpointTest : LanguageServerTestBase
+    {
+        public MonitorProjectConfigurationFilePathEndpointTest()
+        {
+            DirectoryPathResolver = Mock.Of<WorkspaceDirectoryPathResolver>(resolver => resolver.Resolve() == "C:/dir");
+        }
+
+        private WorkspaceDirectoryPathResolver DirectoryPathResolver { get; }
+
+        [Fact]
+        public async Task Handle_Disposed_Noops()
+        {
+            // Arrange
+            var directoryPathResolver = new Mock<WorkspaceDirectoryPathResolver>();
+            directoryPathResolver.Setup(resolver => resolver.Resolve())
+                .Throws<XunitException>();
+            var configurationFileEndpoint = new MonitorProjectConfigurationFilePathEndpoint(
+                Dispatcher,
+                FilePathNormalizer,
+                directoryPathResolver.Object,
+                Enumerable.Empty<IProjectConfigurationFileChangeListener>());
+            configurationFileEndpoint.Dispose();
+            var request = new MonitorProjectConfigurationFilePathParams()
+            {
+                ProjectFilePath = "C:/dir/project.csproj",
+                ConfigurationFilePath = "C:/dir/obj/Debug/project.razor.json",
+            };
+
+            // Act & Assert
+            await configurationFileEndpoint.Handle(request, CancellationToken.None);
```

Comment:
Shouldn't we check the result if we're seeing if it "noops"?

Reply:
Our directory path resolver throws so this will explode if it doesn't no-op. The handle method always returns the same thing whether it no-op'd or not